src/services/Sso.php

```diff
@@ -0,0 +1,186 @@
+ * Single Sign On Cloud Based plugin for CraftCMS
+ * @link      https://websitetoolbox.com/
+ * @copyright Copyright (c) 2020 Website Toolbox
+ */
+namespace websitetoolbox\websitetoolboxforum\services;
+use websitetoolbox\websitetoolboxforum\models\Settings;
+use websitetoolbox\websitetoolboxforum\Websitetoolboxforum;
+use Craft;
+use craft\base\Component;
+use craft\web\View;
+use craft\services\Config;
+define('WT_API_URL', 'https://api.websitetoolbox.com/dev/api');
+/**
+ * @author    Website Toolbox
+ * @package   Websitetoolboxforum
+ * @since     3.0.0
+ */
+class Sso extends Component
+{    
+     function afterLogin(){  
+          $token = Craft::$app->getSession()->get(Craft::$app->getUser()->tokenParam); 
+          if($token){ 
+               $forumApiKey = Craft::$app->getProjectConfig()->get('plugins.websitetoolboxforum.settings.forumApiKey');
+               $forumUrl    = Craft::$app->getProjectConfig()->get('plugins.websitetoolboxforum.settings.forumUrl');
+               if($forumApiKey){ 
+                    $myUserQuery  = \craft\elements\User::find();
+                    $userEmail    = Craft::$app->getUser()->getIdentity()->email;
+                    $userId       = Craft::$app->getUser()->getIdentity()->id;
+                    $userName     = Craft::$app->getUser()->getIdentity()->username;
+                    $RequestUrl   = $forumUrl."/register/setauthtoken";
+                    $postData     = array('type'=>'json','apikey' => $forumApiKey, 'user' => $userName,'email'=>$userEmail,'externalUserid'=>$userId);
+                    $result       = Websitetoolboxforum::getInstance()->sso->sendApiRequest('POST',$RequestUrl,$postData,'json'); 
+                    setcookie("forumLogoutToken", $result->authtoken, time() + 3600,"/");
+                    setcookie("forumLoginUserid", $result->userid, time() + 3600,"/");
+              }
+          }         
+     }
+     function afterUserCreate($userId){ 
+        $forumUrl     = Craft::$app->getProjectConfig()->get('plugins.websitetoolboxforum.settings.forumUrl',false);
+        $forumApiKey  = Craft::$app->getProjectConfig()->get('plugins.websitetoolboxforum.settings.forumApiKey',false); 
+        $postData     = array( 
+                          'type'=>'json',
+                          'apikey'          => $forumApiKey,
+                          'member'          => $_POST['username'],
+                          'externalUserid'  => $userId, 
+                          'email'           => $_POST['email']);
+        if(isset($_POST['firstName'])){
+           $postData['name']  =  $_POST['firstName'];
+        }
+        if(isset($_POST['firstName'])){
+           $postData['name'] .=  " ".$_POST['lastName'];
+        }        
+        $RequestUrl           = $forumUrl . "/register/create_account/";
+        $result               = Websitetoolboxforum::getInstance()->sso->sendApiRequest('POST',$RequestUrl,$postData,'json');          
```

Comment:
What is the user of $result. Because this function didn't return anything.

Reply:
done